bin/report_component_problems.py

```
@@ -68,6 +71,27 @@ def check_component(component: Component) -> str | None:
                 if bool(translation_dict.get("defaultValue")):
                     return "defaultValue has a translation"
 
+        case {"type": "map"}:
+            default_value = component.get("defaultValue")
+            if default_value is None:
+                # No defaultValue is a-okay
+                return None
+
+            try:
+                if isinstance(
+                    default_value, dict
+                ) and GeoJsonGeometryPolymorphicSerializer(data=default_value).is_valid(
+                    raise_exception=True
+                ):
+                    # This object has a valid Geometry shape
+                    return None
+            except ValidationError:
+                return f"Default value '{default_value}' is not valid."
+
+            # Any other value, that doesn't adhere to the Geometry shape, is invalid
+            if default_value is not None:
+                return f"Default value '{default_value}' is not valid."
```

```suggestion
            if isinstance(
                default_value, dict
            ) and GeoJsonGeometryPolymorphicSerializer(data=default_value).is_valid()
                # This object has a valid Geometry shape
                return None
            elif default_value is not None:
                return f"Default value '{default_value}' is not valid."
```

---

Won't this just break if a map config isn't set right? I mean, it will report some sort of problem (i.e. the report breaks because component X), but it's not very nice.. (and if there are multiple components with the same problem, you will have to `run report` > `fix component` > `run report` > `fix component` > ...)

---

yeah there's a branch missing for when the serializer is not valid, so we should capture the result of `is_valid`.

I generally don't like the exception-based flow with serializers if you immediately catch it again, you can just use the boolean return value of `is_valid()` instead

---

Ahh i see the problem..

Yeah but the serializer will still throw an error if using `is_valid()`. For this map defaultValue: 

```
{
    "type": "Point",
    "coordinates": [
      4,
      4,
      5
    ]
  }
```

It throws this error:

```
KeyError: "`GeoJsonGeometryPolymorphicSerializer.serializer_mapping` is missing a corresponding serializer for the `<class 'rest_framework.fields.empty'>` key"
```

So i think we need to throw the error via the `is_valid(raise_exception=True)` and then catch it, right? And the message should be something that reflects the issue of the serializer (i.e. the validation error message)

---

Further discussed via slack, seems to be an issues originating from https://github.com/maykinmedia/drf-polymorphic/blob/816653385c54cb8e28dfaa34eec8ffa1dbb1e8b1/drf_polymorphic/serializers.py#L124.
Decided to throw and catch the error here, and pass the Validation message to the error messages list